mondrian/src/main/java/mondrian/rolap/SqlStatement.java

```diff
@@ -433,17 +435,47 @@ public Object get() throws SQLException {
                 .getMondrianConnection()
                 .getSchema();
 
+            Dialect dialect = getDialect(schema);
+
             if (suggestedType != null) {
                 types.add(suggestedType);
-            } else if (schema != null && schema.getDialect() != null) {
-                types.add(schema.getDialect().getType(metaData, i));
+            } else if (dialect != null) {
+                types.add(dialect.getType(metaData, i));
             } else {
                 types.add(Type.OBJECT);
             }
         }
         return types;
     }
 
+    /**
+     * Retrieves dialect from schema or attempts to create it
+     * in case it is null
+     *
+     * @param schema rolap schema
+     * @return database dialect
+     */
+    protected Dialect getDialect(RolapSchema schema) {
+        Dialect dialect = null;
+        try {
```

Comment:
@mkambol I wanted to cover up cases where connection or datasource are null (or not valid)
and fall back to type.object instead of throwing exception.